crates/driver/src/tests/setup/mod.rs

```diff
@@ -428,22 +558,40 @@ pub fn ab_pool() -> Pool {
     Pool {
         token_a: "A",
         token_b: "B",
-        amount_a: DEFAULT_POOL_AMOUNT_A.into(),
-        amount_b: DEFAULT_POOL_AMOUNT_B.into(),
+        amount_a: DEFAULT_POOL_AMOUNT_A.to_wei(),
+        amount_b: DEFAULT_POOL_AMOUNT_B.to_wei(),
+        liquidity_provider: LiquidityProvider::Amm,
+    }
+}
+
+pub fn ab_pmm_pool(quote: &OrderQuote) -> Pool {
+    Pool {
+        liquidity_provider: LiquidityProvider::Pmm,
```

Comment:
Sounds like PMM should hold onto the quote instead of requiring it on the order? This should be able to let you get rid of the branch that panics below.

Reply:
Initially, just wanted to add flexibility to use a quote for each order. Currently, don't see if it is required anymore. Moved to the PMM.